nixos/modules/services/web-servers/nginx/vhost-options.nix

```diff
@@ -367,4 +386,69 @@ with lib;
       description = "Declarative location config";
     };
   };
+
+  config = let
+   inherit (global-config.services.nginx) package;
+  in {
+    warnings = optional config.enableSSL
+      ''
+        config.services.nginx.virtualHosts."${name}".enableSSL is deprecated,
+        use config.services.nginx.virtualHosts."${name}".onlySSL instead.
+      '';
+
+    assertions = [
+      (let
+        matchedLocations = filterAttrs (n: v: v.root != null && v.alias != null) config.locations;
+      in{
+        assertion = (builtins.length (lib.attrValues matchedLocations)) > 0;
```

Comment:
The nginx testcase fails with

```
       error:
       Failed assertions:
       - Only one of root or alias can be specified on location .
       - Only one of root or alias can be specified on location .
```

I assume this is because the condition should be `== 0` here?
I'm wondering if it isn't easier to just do `matchedLocations == {}` as assertion here.

Reply:
I changed it for now but I need to re-test it with my config but in the test it seems to work as expected. I have a variety of things but that didn't trigger it for my config.